docs/instrumentation-libraries.md

```diff
@@ -0,0 +1,99 @@
+# Support for OpenTelemetry Instrumentation Libraries
+
+OpenTelemetry instrumentation are implemented using a variety of techniques.
+Sometimes the instrumentation is built directly into the library being
+instrumented, sometimes the instrumentation requires the addition of a separate
+instrumentation library, and other times it is a hybrid of those approaches. This
+project aims to make using these instrumentation approaches easier and more
+automatic for our users. This requires us to both vet and dynamically enable a
+subset of the instrumentation that is available to the OpenTelemetry community,
+while still providing some flexibility to allow users to manually load and enable
+instrumentation that we do not support out of the box.
+
+## Libraries and our instrumentation support plans
+
+The libraries listed here contain our current thoughts about which libraries should
+include out of the box support, and which libraries we decided to not support. If
+you have a library that you would like to consider adding to this list, please
+submit an issue to request its inclusion.
+
+### Application Frameworks
+
+1. ASP.NET
+1. AspNetCore
+1. Windows Workflow Foundation
+1. OWIN
+1. Azure Service Fabric
+
+### Databases
+
+#### General database frameworks
+
+1. ADO.NET
+1. EntityFramework
+1. Dapper
+1. System.Data.LINQ
+1. Microsoft.Practices.EnterpriseLibrary.Data
+1. EntityFrameworkCore
+
+#### Specific Databases
+
+1. MySQL
```

Comment:
It's not clear what does "MySQL" mean. Is it a particular client library, or any libraries that interact with any version of MySQL?

Reply:
Before we can clarify that, I think we'll need to decide what supporting a particular database means. Does that mean we support specific driver versions, and we have tests focused on validating support of those driver versions? Or, does that mean we have tests, using specific drivers in addition to specific databases? In the short-term it may be easier to focus on just the driver-side of things, but in the long term we may need to also have some emphasis on testing and listing compatibility for databases too.

Reply:
+1

Reply:
We are instrumenting libraries, therefore I think we should say that we instrument an ADO.NET driver library. We can always add notes on what the instrumented library is for. For example, it may be not obvious that `Npgsql` is a driver for PostgreSQL.

Reply:
> We are instrumenting libraries, therefore I think we should say that we instrument an ADO.NET driver library. We can always add notes on what the instrumented library is for. For example, it may be not obvious that `Npgsql` is a driver for PostgreSQL.

You are right, when I convert this section to use a table, I can make the driver/library the focus, but I think it will be important to call out the databases we validate against, and a note about the relevant databases for that driver/library. When supporting instrumentation, I usually get asked the question "do you support this database?" instead of "do you support this driver?". I do not know if that is a result of the original question changing by the time it reaches engineering, or if it is related to New Relic's explain plan support (which requires database and driver specific variations).

Reply:
Do the changes in [ea07f30](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/931/commits/ea07f30c16de3e510b096e5484d804b4028ed4d9) make things a little clearer?

Reply:
Awesome